File: src/Client/IChannel.cs
using System;
using System.Threading.Tasks;
using AppBoxCore;

namespace AppBoxClient;

public interface IChannel
{
    string SessionName { get; }
    
    Guid LeafOrgUnitId { get; }
    
    Task Login(string user, string password, string? external);

    Task Logout();

    Task<object?> Invoke(string service, object?[]? args, EntityFactory[]? entityFactories);
}

File: src/Dev.Web/src/AppBoxClient/Channel/IChannel.ts
import {EntityFactory} from "@/AppBoxCore";

export interface IChannel {

    get SessionId(): number | null;

    get Name(): string | null;

    /** 登录至服务端 */
    Login(user: string, password: string, external?: string): Promise<void>;

    Logout(): Promise<void>;

    Invoke<T>(service: string, args?: any[], entityFactories?: Map<bigint, EntityFactory>): Promise<T>;
}